lib/chat_channel_membership_manager.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+class DiscourseChat::ChatChannelMembershipManager
+  def self.all_for_user(user)
+    UserChatChannelMembership.where(user: user)
+  end
+
+  def self.find_for_user(user:, channel_id: nil, channel: nil, following: nil, initialize: false)
+    ensure_channel_or_id_provided!(channel_id, channel)
+    params = { user_id: user.id, chat_channel_id: channel_id || channel.id }
+    params[:following] = following if following.present?
+
+    if initialize
+      UserChatChannelMembership.find_or_initialize_by(params)
+    else
+      UserChatChannelMembership.includes(:user, :chat_channel).find_by(params)
+    end
+  end
+
+  def self.follow_channel(user:, channel_id: nil, channel: nil)
+    ensure_channel_or_id_provided!(channel_id, channel)
+    membership =
+      find_for_user(user: user, channel_id: channel_id, channel: channel, initialize: true)
+
+    ActiveRecord::Base.transaction do
+      if !membership.following
+        membership.following = true
+        membership.save!
+        recalculate_user_count!(channel_id || channel.id)
+      end
+    end
+
+    membership
+  end
+
+  def self.unfollow_channel(user:, channel_id: nil, channel: nil)
```

Comment:
where is the `channel_id`  signature used? If we could I would prefer a signature like this one: `unfollow(acting_user:, channel:)`

Reply:
Yeah not so sure about this channel_id thing I did, I think in all cases we've had the channel already, so will get rid of this.